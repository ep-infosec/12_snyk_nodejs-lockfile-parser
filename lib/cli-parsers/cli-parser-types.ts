export interface FormattedCliOutput {
  topLevelDeps: string[];
  dependencies: FlatDependenciesMap;
}

export type FlatDependenciesMap = Map<string, string[]>;

export type YarnInfoOutput = Array<{
  value: string;
  // `children` has many possible keys in the objects but
  // these are the types we care about though
  children: {
    Version: string;
    Dependents?: string[];
    Dependencies?: {
      descriptor: string;
      locator: string;
    }[];
  };
}>;

export interface YarnListTree {
  name: string;
  children: { name: string; color: string; shadow: boolean }[];
  hint: string | null;
  color: string | null;
  depth: number;
}
