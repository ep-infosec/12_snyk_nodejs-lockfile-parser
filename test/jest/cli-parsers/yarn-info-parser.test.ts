import * as fs from 'fs';
import { parseYarnInfoOutput } from '../../../lib/cli-parsers/yarn-info-parser';

describe('parseYarnListOutput', () => {
  it('should return a correct flat map of the dependencies', () => {
    const yarnInfoOutput = fs.readFileSync(
      `${__dirname}/fixtures/yarn-info-output.txt`,
      'utf8',
    );
    const result = parseYarnInfoOutput(yarnInfoOutput);
    expect(result).toMatchSnapshot();
  });
});
