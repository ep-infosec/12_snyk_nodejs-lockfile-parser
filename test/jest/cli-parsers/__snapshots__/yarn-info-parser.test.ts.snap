// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parseYarnListOutput should return a correct flat map of the dependencies 1`] = `
Object {
  "dependencies": Map {
    "accepts@npm:1.3.7" => Array [
      "mime-types@npm:2.1.31",
      "negotiator@npm:0.6.2",
    ],
    "mime-db@npm:1.48.0" => Array [],
    "mime-types@npm:2.1.31" => Array [
      "mime-db@npm:1.48.0",
    ],
    "negotiator@npm:0.6.2" => Array [],
  },
  "topLevelDeps": Array [
    "accepts@npm:1.3.7",
  ],
}
`;
